src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -25,9 +33,25 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
         public string? ClientName { get; }
 
         public RequestContext(ClientCapabilities clientCapabilities, string? clientName)
+            : this(null, null, clientCapabilities, clientName)
         {
+        }
+
+        internal RequestContext(Solution? solution, Action<Solution>? solutionUpdater, ClientCapabilities clientCapabilities, string? clientName)
```

Comment:
optional stuff before non-optional seems weird.

Reply:
Will clean up in a future PR, there is more to add here